src/main/java/org/spongepowered/common/event/tracking/PhaseTracker.java

```diff
@@ -224,6 +224,7 @@ public void completePhase(IPhaseState<?> prevState) {
         } catch (Exception e) {
             this.printMessageWithCaughtException("Exception Post Dispatching Phase", "Something happened when trying to post dispatch state", state, context, e);
         }
+        this.checkPhaseContextProcessed(state, context);
```

Comment:
Maybe this could be flagged for validation? If we have fixed all the bugs where the check is no longer necessary, it'd be extra processing that wouldn't be necessary.

Reply:
I don't see this check adding much overhead at all. I think it's useful to always have on, in order to catch any regressions that might occur (like the TNT bug).